<launch>
	<arg name="base_frame" default="robot_0/base_link" />
	<!-- <arg name="odom_frame" default="robot_0/odom_combined" /> -->
	<arg name="odom_frame" default="robot_0/odom" />
	<arg name="map_frame" default="map" />
	<arg name="use_deadman" default="false" />
	<arg name="person_frame" default="person_raw" />


	<param name="/use_sim_time" value="true"/>
	<arg name="move_base_goal_frequency" default="0.5" />

	<node pkg="stage_ros" type="stageros" name="stageros_node" args="$(find blob_follower)/worlds/simple.world">
		<remap from="/robot_0/blob" to="/robot_0/blob_0" />
		<remap from="/robot_0/cmd_vel" to="/cmd_vel" />
	</node>


	<!-- nodelets and its manager for cmd_vel mux -->

	<!-- <include file="$(find blob_follower)/launch/ekf.launch" /> -->

	<!-- <node pkg="slam_karto" type="slam_karto" name="slam_karto" output="screen"> -->

	<node name="map_server" pkg="map_server" type="map_server"
	args="$(find blob_follower)/worlds/eight.yaml" output="screen"/>
	<node pkg="tf" type="static_transform_publisher" name="link1_broadcaster" args="10 10 0 0 0 0 -1 map $(arg odom_frame) 100" />
	<include file="$(find blob_follower)/launch/amcl.launch">
		<remap from="scan" to="robot_0/base_scan_1"/>
		<param name="odom_frame_id" value="$(arg odom_frame)"/>
		<param name="base_frame_id" value="$(arg base_frame)"/>
		<param name="global_frame_id" value="map"/>
	</include>

	<!-- <node pkg="gmapping" type="slam_gmapping" name="slam_gmapping" output="screen">
		<remap from="scan" to="robot_0/base_scan_1"/>
		<param name="odom_frame" value="$(arg odom_frame)"/>
		<param name="base_frame" value="$(arg base_frame)"/>
		<param name="map_frame" value="map"/>
	    <param name="map_update_interval" value="0.025"/>
	    <param name="particles" value="20" />
	    <param name="delta" value="0.05"/>
	    <param name="maxRange" value="7"/>
	    <param name="xmin" value="-10"/>
	    <param name="ymin" value="-10"/>
	    <param name="xmax" value="10"/>
	    <param name="ymax" value="10"/>
	    <rosparam command="load" file="$(find blob_follower)/config_nav/mapper_params.yaml" />
	</node>
 -->
	<include file="$(find blob_follower)/config_nav/move_base.xml" />

	<node pkg="rviz" type="rviz" name="rviz" output="screen" />
		  <!--args="-d $(find blob_follower)/config/simulation.rviz"/>-->

	<!--<node pkg="joy" type="joy_node" name="joy_node" output="screen">-->
		<!--<remap from="/joy" to="/teleop/joy" />-->
		<!--<param name="dev" type="string" value="/dev/input/js0" />-->
		<!--&lt;!&ndash; <param name="dev" type="string" value="/dev/input/js1" /> &ndash;&gt;-->
		<!--<param name="autorepeat_rate" value="10" />-->
	<!--</node>-->

	<!--<node pkg="teleop_twist_joy" type="teleop_node" name="teleop_node">-->
		<!--<remap from="joy" to="teleop/joy" />-->
		<!--<remap from="cmd_vel" to="robot_1/cmd_vel" />-->
		<!--<param name="scale_linear" value="0.75" />-->
		<!--<param name="scale_angular" value="0.5" />-->
	<!--</node>-->
</launch>